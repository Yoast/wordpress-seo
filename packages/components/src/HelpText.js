/* External dependencies */
import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

/* Yoast dependencies */
import { colors } from "@yoast/style-guide";

const YoastHelpText = styled.p`
	color: ${ props => props.textColor };
	font-size: ${ props => props.textFontSize };
	margin-top: 0;
`;

/**
 * Returns the HelpText component.
 *
 * @param {Object} props              Component props.
 * @param {string} props.text         The help text.
 * @param {string} props.textColor    The CSS textColor of the paragraph text.
 * @param {string} props.textFontSize The CSS font-size of the paragraph text.
 *
 * @returns {ReactElement} HelpText component.
 */
export default class HelpText extends PureComponent {
	/**
	 * Renders a help text component.
	 *
	 * @returns {ReactElement} The rendered help text component.
	 */
	render() {
		const { children, textColor, textFontSize } = this.props;

		return (
			<YoastHelpText textColor={ textColor } textFontSize={ textFontSize }>
				{ children }
			</YoastHelpText>
		);
	}
}

/**
 * React prop type for the help text.
 *
 * Use this in your components to pass along the text.
 */
export const helpTextPropType = {
	children: PropTypes.oneOfType( [
		PropTypes.string,
		PropTypes.array,
	] ),
	textColor: PropTypes.string,
	textFontSize: PropTypes.string,
};

HelpText.propTypes = {
	...helpTextPropType,
	children: helpTextPropType.children.isRequired,
};

HelpText.defaultProps = {
	textColor: colors.$color_help_text,
};
