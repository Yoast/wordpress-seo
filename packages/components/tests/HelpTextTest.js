/* External dependencies */
import React from "react";
import renderer from "react-test-renderer";

/* Internal dependencies */
import HelpText from "../src/HelpText";

describe( "HelpText", () => {
	it( "matches the snapshot by default", () => {
		const component = renderer.create(
			<HelpText>
				{ "Some help text." }
			</HelpText>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "matches the snapshot when an array is provided as text", () => {
		const component = renderer.create(
			<HelpText>
				{ [ "Text ", <a key="1" href="https://www.example.org">with a link</a>, " in the middle." ] }
			</HelpText>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
