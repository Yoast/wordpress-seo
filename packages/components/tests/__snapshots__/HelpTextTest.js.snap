// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpText matches the snapshot by default 1`] = `
.c0 {
  color: #767676;
  margin-top: 0;
}

<p
  className="c0"
>
  Some help text.
</p>
`;

exports[`HelpText matches the snapshot when an array is provided as text 1`] = `
.c0 {
  color: #767676;
  margin-top: 0;
}

<p
  className="c0"
>
  Text 
  <a
    href="https://www.example.org"
  >
    with a link
  </a>
   in the middle.
</p>
`;
